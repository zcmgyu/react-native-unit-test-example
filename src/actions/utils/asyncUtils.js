/**
 * @param baseName action base name
 * @return {object} an object with the three async action names
 */
export const asyncActionNames = baseName => ({
  failure: `${baseName}_FAILURE`,
  request: `${baseName}_REQUEST`,
  success: `${baseName}_SUCCESS`
});

/**
 * @param actionName {object}
 * @return {object} the three async object actions
 */
export const buildAsyncActions = actionName => ({
  request: () => ({
    type: actionName.request
  }),
  failure: error => ({
    type: actionName.failure,
    error
  }),
  success: data => ({
    type: actionName.success,
    data
  })
});
