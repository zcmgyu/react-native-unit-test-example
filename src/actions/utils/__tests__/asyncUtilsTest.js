import * as utils from '../asyncUtils.js';

it('returns the proper action names for TEST_API on asyncActionNames', () => {
  expect(utils.asyncActionNames('TEST_API')).toMatchSnapshot();
});

it('builds a proper object for async actions on buildAsyncActions', () => {
  const TEST_API = utils.asyncActionNames('TEST_API');
  const receiveTestData = utils.buildAsyncActions(TEST_API);

  const data = [{ id: 1, name: 'Fake' }];
  const error = new Error('Look ma, I am an error!');

  expect(receiveTestData.failure(error)).toMatchSnapshot();
  expect(receiveTestData.request()).toMatchSnapshot();
  expect(receiveTestData.success(data)).toMatchSnapshot();
});
