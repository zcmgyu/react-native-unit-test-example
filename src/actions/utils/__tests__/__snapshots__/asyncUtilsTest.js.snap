// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`builds a proper object for async actions on buildAsyncActions 1`] = `
Object {
  "error": [Error: Look ma, I am an error!],
  "type": "TEST_API_FAILURE",
}
`;

exports[`builds a proper object for async actions on buildAsyncActions 2`] = `
Object {
  "type": "TEST_API_REQUEST",
}
`;

exports[`builds a proper object for async actions on buildAsyncActions 3`] = `
Object {
  "data": Array [
    Object {
      "id": 1,
      "name": "Fake",
    },
  ],
  "type": "TEST_API_SUCCESS",
}
`;
exports[`returns the proper action names for TEST_API on asyncActionNames 1`] = `
Object {
  "failure": "TEST_API_FAILURE",
  "request": "TEST_API_REQUEST",
  "success": "TEST_API_SUCCESS",

}
`;
